packages/story-editor/src/app/mediaUpload/provider.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+
+import { useRef } from '@web-stories-wp/react';
+import PropTypes from 'prop-types';
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '..';
+import Context from './context';
+
+function OpenMediaUploadHandler({ onOpenMediaUpload, ...props }) {
+  /**
+   * `open` needs to be stored in a ref so that it does not cause endless
+   * re-renders for any children.
+   */
+  const openRef = useRef(open);
```

Comment:
should this be:
```
const openRef = useRef(onOpenMediaUpload);
```

Reply:
I'm not following how this as a ref like this changes the behavior at all vs just passing the `onOpenMediaUpload` prop right through the provider. Mind breaking it down for me?

Reply:
oops yeah, good catch

We really should pass the ref through, not the value of the ref. Monday brain 😮‍💨

Reply:
I think this would work?
```js
function OpenMediaUploadHandler({ onOpenMediaUpload, ...props }) {
  /**
   * `open` needs to be stored in a ref so that it does not cause endless
   * re-renders for any children.
   */
  const openRef = useRef(onOpenMediaUpload);
  openRef.current = onOpenMediaUpload;

  const handleOpenMediaUpload = useCallback((ev) => openRef.current(ev), []);

  return (
    <Context.Provider
      {...props}
      value={{ onOpenMediaUpload: handleOpenMediaUpload }}
    />
  );
}
OpenMediaUploadHandler.propTypes = {
  onOpenMediaUpload: PropTypes.func.isRequired,
};

```

Reply:
But the idea is that if the `open` function that's taken in from the render prop changes, then it's added to the ref. Hopefully this wouldn't cause an extra re-render every time something above this re-renders since the `onOpenMediaUpload` function reference never changes